tsl/src/continuous_aggs/insert.c

```diff
@@ -471,13 +476,20 @@ get_lowest_invalidated_time_for_hypertable(Oid hypertable_relid)
 		.lockmode = AccessShareLock,
 		.scandirection = ForwardScanDirection,
 		.result_mctx = NULL,
+
+		/* We need to define a custom snapshot for this scan. The default snapshot (SNAPSHOT_SELF)
+		   reads data of all committed transactions, even if they have started after our scan. If a
+		   parallel session updates the scanned value and commits during a scan, we end up in a
+		   situation where we see the old and the new value. This causes ts_scanner_scan_one() to
+		   fail. */
+		.snapshot = GetTransactionSnapshot(),
```

Comment:
Wondering if we should do the same in https://github.com/timescale/timescaledb/blob/main/tsl/src/continuous_aggs/invalidation_threshold.c#L163

Reply:
Yes, we should do it. However, this is already done in this pending PR: https://github.com/timescale/timescaledb/pull/6672. There, we use the latest snapshot to see parallel updates.